src/Storages/StorageURLCluster.cpp

```diff
@@ -76,7 +76,7 @@ StorageURLCluster::StorageURLCluster(
 
     storage_metadata.setConstraints(constraints_);
     setInMemoryMetadata(storage_metadata);
-    setVirtuals(VirtualColumnUtils::getVirtualsForFileLikeStorage(storage_metadata.getColumns()));
+    setVirtuals(VirtualColumnUtils::getVirtualsForFileLikeStorage(storage_metadata.getColumns(), context));
```

Comment:
Let's also support hive partitioning for `urlCluster`. We can reuse `getSampleURI` function (we can move it to `StrorageURL` header or make it a static method of `IStorageURLBase`)